fix node-libcurl imports

dimitropoulos opened the pull request:
Back in the flow days, we needed to reach into these files to get at the enum, but now we can import directly.  It's a fairly strong anti-pattern to reach into `/dist` like this on npms because there are no guarantees about what will be there in the future, which greatly increases the surface-area for breakage in our own code.

dimitropoulos committed:
fix node-libcurl imports

develohpanda reviewed (approved):
Result of having flow and not being able to import those directly (I think?)

DMarby reviewed (approved):
Looks reasonable, seems like it's breaking the tests however?

dimitropoulos committed:
removes unnecessary function wrapper

and anyway, `number` is not the correct type (which is what motivated this change in the first place), `CurlFeature` is.

dimitropoulos committed:
updates type for setOpt

see https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgOIBs4GcvIN4BQyxyWwAXhAFzIDkWAtnOurcgD50MQAmwArgzadamKAHMItANwEAvgQKhIsRCgDCAe00BrYCkIlkYAJ4AHanSz9xcKMLoIAFpoSbMkdU+BmZ8xQD0AcjK0PBIyJoAbtDomnA8WErgYWrIAIIIYMCaIPhEJBBwYACMABSaZtm5NLQAssDoOrQANMjimDg0GNhYAJQ0UZrAPLJGRaUVVTkgtVq6+lityG4LlvN6EAPIQyOyCgRuIFhgyIjVsxlZM8gAvPnjxeWVF208xXB9+MhByGZQmgARugIAwaKYLCFcCBNKdQDBoFBeGcQDxkABJZBOOAxZAw5AAdzgJmMmmM5hQwFOsKc0AJwCwEAKRmIRyw7ggADo4uIyrRzjNORMSss8JFprk3h9kHI+syZf5DrkTshhQB1KBwMwWKA0TIXADatGFtAAundkFNXsh3mBPncAHzfX7-IEgsHkyEMvGwkIgBFQJFouCojFYnEofFEklgMkQynUsC0qD0xnytkc7maXnGp6i8XW21wGV9MaFJ4arU6q0zKV20uKgi-JHifhiZAwfgga65SIxKBxBJJTvdi6q4oAJhrNToDSayw6vW6nX6g2GowII57eQmU5eMzm2k2Szaq02NA2+m2uw3W7Hu+nIDr9rFLoBwNB4IpUJ9cP9iOREM0UxbFcSjYlSU9BNIiTOkGRQMpvQAAxDEwkOMWlIhAdASX3XIAH5iHSZAuxuBkQFoU43EDCAslVKlk05OUjAzEEsxzFs2zsDsu23fM8KfG1pVlRU2VOXdK21aAvwsTQYHHMAJwtR9ny+W4nVfYJXQ-D14x-GE-wDIMUWA8MwLJaNIL0qkYOTVMmRY5VMx5Pld2QSSdVoBtHkUlShPrfZAmCWDGUJTQoB0NpsFVAAPCwsl4KgmyCFLUrS9KMsyrKUoIAVciFJ4+TnZo2jFMhKFqRhmFYEtZHVTUpKgIrGhK75yssegmBYNhZTqydmvnUrSAoDqqu62qCAkhrq3qFr83ayqupq3rFDykACsmWhLwgE9vnjWpnFcdxiggLwfB6uV6qraA+W23axX2xwXDcDxTu8XwS0m-qtqPRZ80e2hDpek6zo+kSpuupqfrWe6oIO57js8d6LqCjCdpQAAqHgyQMjGwoiqLcAgOLaMgHgkuyymqepoJkuQAABMAsAAWmJ+KwFZwNwuQQEEg7YAoBVOxW24cBcu3DbymhzZ-opeGjte0GUd+RmWbZ0nOYBKAeb5mABaFiRBAgMWrsa27fqkIaAaBxG3vOz6VaZ1mSayTXud5tE9cF05haNk3vu22WLHl4GkftkTHbVl2OcRd3df1n3DdFsAvsUjybul-Qg46m3FeRz66dV532bd7WPdIWjcmDJPjZTtbJfNmHs5D23meccO5Uj4uNdjsu+cZI5q5F2vU5KdOobu5unoVk62-ziPgiL9XXd7nW0QHquzhr-3fMznap8BhHXrnjvC6d5eY65vv18r0NfeT1OJ3HxvjwP3PZ-bsG5SAA

to see why `any` is required, for now (at least)

dimitropoulos committed:
updates the getBodyBuffer parse calls now that the types are working

dimitropoulos committed:
return to prior approach, this time with a warning

dimitropoulos commented:
@develohpanda @DMarby turns out there was a reason after all.  tried to be decently good about atomic commits if you wanna see what happened, but I was (at least) able to get rid of the call into /dist from the non-test, but the test needs to keep it otherwise you get something like:

```
  ● Test suite failed to run

    The module '/home/dimitri/src/github.com/Kong/insomnia/packages/insomnia-app/node_modules/node-libcurl/lib/binding/node_libcurl.node'
    was compiled against a different Node.js version using NODE_MODULE_VERSION 80.
    This version of Node.js requires NODE_MODULE_VERSION 72. Please try re-compiling or re-installing
    the module (for instance, using `npm rebuild` or `npm install`).

      at Runtime._loadModule (../node_modules/jest-runtime/build/index.js:893:29)
      at Object.<anonymous> (../node_modules/node-libcurl/lib/Easy.ts:10:44)

```

develohpanda commented on packages/insomnia-app/app/common/render.ts:
```diff
@@ -419,9 +421,10 @@ export async function getRenderedGrpcRequestMessage(
 export async function getRenderedRequestAndContext(
   request: Request,
-  environmentId: string | null,
+  environmentId?: string | null,
```
Checked this and it looks like an acceptable change. Can now remove this fallback: https://github.com/Kong/insomnia/blob/1271f5a925c956452165fa28384c6dfb84f6fdec/packages/insomnia-app/app/network/network.ts#L965

dimitropoulos replied:
![Screenshot_20210517_091947](https://user-images.githubusercontent.com/15232461/118495262-08721580-b6f1-11eb-8e9a-f4037c0528aa.png)

unfortunately not, `undefined` is not covariant with the environmentId arg of getRenderContext.


Making this argument optional was added to fix this error:
![Screenshot_20210517_092504](https://user-images.githubusercontent.com/15232461/118496053-c5fd0880-b6f1-11eb-8cfa-6f76bbe49968.png)

Which appears to be genuine (every call in `request.test.ts` only passes the first argument, as well as every call in `har.test.ts`).

develohpanda commented on packages/insomnia-app/app/network/network.ts:
```diff
@@ -461,7 +446,7 @@ export async function _actuallySend(
       if (renderedRequest.settingSendCookies) {
         // Tell Curl to store cookies that it receives. This is only important if we receive
         // a cookie on a redirect that needs to be sent on the next request in the chain.
-        curl.setOpt(Curl.option.COOKIEFILE, '');
```
This is what happens when the helper/wrapper and the parent function carry the same name 😆 good catch

dimitropoulos replied:
I chuckled, too.  We can try as hard as we want, but in the end, this is a thing made by humans.  This is a hard mistake to avoid.

develohpanda reviewed (approved):
Changes look good to me

dimitropoulos committed:
Merge branch 'develop' into node-libcurl-types
